tools/pipelines/templates/build-npm-package.yml

```diff
@@ -152,10 +152,14 @@ stages:
                 publish=${{ variables.publish }}
                 canRelease=${{ variables.canRelease }}
                 publishNonScopedPackages=${{ variables.publishNonScopedPackages }}
-                targetBranchName = $(System.PullRequest.TargetBranch)
 
                 release=$(release)"
 
+              # Assign target branch
+              if [[ ${{ variables['Build.Reason'] }} == "PullRequest" ]]; then
+                echo "targetBranchName = $(System.PullRequest.TargetBranch)"
+              fi
+
```

Comment:
Ah, I see the reason for this in https://github.com/microsoft/FluidFramework/pull/12352. I suggest we go the way I proposed in [AB1862](https://dev.azure.com/fluidframework/internal/_workitems/edit/1862/) and make `targetBranchName` an actual variable in the job (define it conditionally in lines 112-114 above). This version of the change will fix the original issue I saw but note that `$(System.PullRequest.TargetBranch)` is also used in lines 390 and 396, so those could still fail; other conditions around them might make it so in practice it never fails, but it feels more correct to completely wrap any use of `$(System.PullRequest.TargetBranch)` with the right condition.

Reply:
Related, I just opened https://github.com/microsoft/FluidFramework/pull/12676 to remove line 396 which seems like it might have been a mistake from a copy paste. If that PR gets merged before this you might run into a merge conflict when making the change I suggested above. If you do, know that that line can just go away.

Reply:
> Ah, I see the reason for this in #12352. I suggest we go the way I proposed in [AB1862](https://dev.azure.com/fluidframework/internal/_workitems/edit/1862/) and make `targetBranchName` an actual variable in the job (define it conditionally in lines 112-114 above). This version of the change will fix the original issue I saw but note that `$(System.PullRequest.TargetBranch)` is also used in lines 390 and 396, so those could still fail; other conditions around them might make it so in practice it never fails, but it feels more correct to completely wrap any use of `$(System.PullRequest.TargetBranch)` with the right condition.

Updated. And still print the `targetBranchName` in `Parameter` step (with condition), I think we can keep that.